### ISSUE TITLE
_⚠️ Potential issue_ sliding window attention mask

### ISSUE THREAD

**Issue by kirilklein** [FGA-DIKU/EHR#251]
_⚠️ Potential issue_

**Causal masking helper can receive `None` or non‑floating dtypes**

1. `sliding_window_mask` can legitimately be `None` depending on the attention type; calling `make_attention_causal(None)` will crash.  

```diff
-        if self.is_causal:
-            global_attention_mask = make_attention_causal(global_attention_mask)
-            sliding_window_mask = make_attention_causal(sliding_window_mask)
+        if self.is_causal:
+            if global_attention_mask is not None:
+                global_attention_mask = make_attention_causal(global_attention_mask)
+            if sliding_window_mask is not None:
+                sliding_window_mask = make_attention_causal(sliding_window_mask)
```

_Originally posted by @coderabbitai[bot] in https://github.com/kirilklein/PHAIR_EHR/pull/60#discussion_r2055824629_

**Comment by kirilklein** [comment]
_update_attention_mask does always return not-none sliding window mask
